compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -302,6 +303,10 @@ public void visit(BLangBlockStmt blockNode) {
         blockNode.stmts.forEach(statement -> analyzeNode(statement, blockEnv));
     }
 
+    @Override
+    public void visit(BLangLetExpression letExpression) {
```

Comment:
Shouldn't this be implemented? 

Does dataflow analysis happen, to identify `i` here is un-initialized?

```ballerina
public function main() {
    int i;
    int b = let int x = 4 + i in 2 * x * i;
}
```

Reply:
+1. Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/21209/commits/ef33eef791e71c704f14386e926d75a8cf047d34

Reply:
Shall we add tests too?

Reply:
Added.